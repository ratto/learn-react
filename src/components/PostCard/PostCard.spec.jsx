import { render, screen } from '@testing-library/react';
import { PostCard } from ".";

const mock = {
    id: 1,
    title: 'Mock title',
    body: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Mauris libero tellus, tristique sit amet viverra id, ultricies sit amet dolor. Vivamus turpis sem, auctor id sollicitudin vitae, cursus in sapien.',
    cover: {
        title: 'image alt',
        url: 'img/mock.png'
    }
}

describe('<PostCard />', () => {
    let postCard;
    beforeAll(() => {
        postCard = (<PostCard {...mock} />);
    });

    it('should render a PostCard with an image.', () => {
        render(postCard);
        const postImage = screen.getByRole('img');

        expect(postImage).toBeInTheDocument();
        expect(postImage).toHaveAttribute('src', mock.cover.url);
        expect(postImage).toHaveAttribute('alt', mock.cover.title);
    });

    it('should render a PostCard with a title and an body text.', () => {
        render(postCard);
        const heading = screen.getByRole('heading', { name: mock.title });
        const bodyText = screen.getByText(mock.body);

        expect(heading).toBeInTheDocument();
        expect(bodyText).toBeInTheDocument();
    });

    it('should match the snapshot taken.', () => {
        const { container } = render(postCard);

        expect(container.firstChild).toMatchSnapshot();
    });
});