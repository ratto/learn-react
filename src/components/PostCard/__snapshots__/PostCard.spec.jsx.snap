// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> should match the snapshot taken. 1`] = `
<div
  class="post"
>
  <img
    alt="image alt"
    src="img/mock.png"
  />
  <div
    class="post-content"
  >
    <h2>
      Mock title
    </h2>
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Mauris libero tellus, tristique sit amet viverra id, ultricies sit amet dolor. Vivamus turpis sem, auctor id sollicitudin vitae, cursus in sapien.
    </p>
  </div>
</div>
`;
